src/scenario/scenario.c

```diff
@@ -459,35 +536,58 @@ void scenario_load_state(buffer *buf, int version)
     scenario.empire.distant_battle_roman_travel_months = buffer_read_u8(buf);
     scenario.empire.distant_battle_enemy_travel_months = buffer_read_u8(buf);
     scenario.open_play_scenario_id = buffer_read_u8(buf);
+
     if (version > SCENARIO_LAST_UNVERSIONED) {
         buffer_read_raw(buf, scenario.empire.custom_name, sizeof(scenario.empire.custom_name));
     }
     buffer_skip(buf, 1);
 
+    if (version <= SCENARIO_LAST_NO_EXTENDED_REQUESTS) {
+        scenario.is_saved = 1;
+    }
+}
+
+void scenario_requests_load_state(buffer *buf, int version)
+{
+    scenario_request_load_state(buf);
```

Comment:
This feels hacky.

I think you could leave the `is_saved = 1` above and directly call `scenario_request_load_state()`.

Reply:
Done. Have to pass 2 buffer pointers to 'scenario_load_state' now though.